### PR TITLE
E2E: New test for select box

### PR DESCRIPTION
**Pull request by 4ydan**
#839
Together with @badnames, we have implemented this new e2e test.

One issue though:
There is currently no real way to confirm how many plants you have selected with the select box, so this test only confirms that the plant name on the bottom left panel is visible.
Open for any other ideas on how to assert/test this.

We would need to wait for the actual fix, to verify if the test behaves.

## Basics

- [x] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

**Comment by markus2330** [comment]
Great job, thank you!

I think we can live with this limitation for now. Is there a good way in Gherkin to tell about such limitations? Maybe simply be more honest to what it does, so instead of "the plant is selected" say "the name of the plant is in the right toolbar" or similar?

I think @badnames can simply push here the fix. It is very good style to first define the behavior of a feature anyway.

**Comment by 4ydan** [comment]
Not sure if we want the fix to be also in this PR.

**Comment by markus2330** [comment]
> Not sure if we want the fix to be also in this PR.

What do you propose? Otherwise we cannot merge this without breaking CI?

**Comment by 4ydan** [comment]
> What do you propose?

I thought we first push just the fix for the select box in a separate PR, but doesnt matter that much I guess.

File: doc/changelog.md [hunk 1]
```diff
@@ -58,12 +58,12 @@ Syntax: `- short text describing the change _(Your Name)_`
...
 - _()_
 - _()_
+- Fix drag select not working and add appropriate e2e regression tests _(4ydan & Moritz)_
 - _()_
 - _()_
 - _()_
 - _()_
-- _()_
-- _()_
+- E2E: New canvas select box test #852 \_(4ydan)\_g
```
Comment (4ydan) [inline doc/changelog.md:66]: And remove this changelog line or the other.

File: e2e/pages/maps/planting.py [hunk 2]
```diff
@@ -120,6 +120,18 @@ def click_on_canvas_middle(self):
...
 
+    def drag_select_box_over_canvas(self):
+        """Drags a select box over the canvas from top left to bottom right"""
+        box = self._canvas.bounding_box()
+        self._page.mouse.move(
+            box["x"] + box["width"] / 2 + 20, box["y"] + box["height"] / 2 + 20
+        )
+        self._page.mouse.down()
+        self._page.mouse.move(
+            box["x"] + box["width"] / 2 - 20, box["y"] + box["height"] / 2 - 20
```
Comment (4ydan) [inline e2e/pages/maps/planting.py:131]: We should remove these magical constants 20.

**Comment by badnames** [comment]
jenkins build please

**Comment by 4ydan** [comment]
This PR is done, but it introduces a new bug.
Also the e2e test is performing a second mouse move for the select box, which is kinda weird, but seems to be the only way to make it work.
Maybe konva works similar like this:
image

**Comment by markus2330** [comment]
@4ydan thx for the careful review! :heart: 

@badnames can you fix this?

**Comment by badnames** [comment]
I investigated this for about one hour, but I did not manage to fix it yet unfortunately.
What I found is that the responsible onMouseUp callback is not invoked by Konva if a plant is selected for placement.
To me, this therefore does not seem to be a new bug.

I think its just that nobody noticed until now, because it only results in the user having to click a second time to end the selection.
Overall this seems to be a low priority issue to be fixed at a later time.

If you want a quick fix, I suggest we disable drag selection completely while placing plants.

**Comment by markus2330** [comment]
@badnames can you please create an issue for that (if it is independent of this PR)?

@tschawax can you take a quick look at the bug fix in this PR?

**Comment by markus2330** [comment]
Thx for the e2e test, fixing the bug and the review, great teamwork :heart:

**Pull request merged**